### PR TITLE
Restore Catalyst unit testing in CI

### PR DESCRIPTION
**Pull request by paulb777**
Turns on most of unit tests for Catalyst
- Disables tests that require the keychain (all of Auth and some of IID and GoogleUtilities)
- Disables the flaky GULNetwork, GULSwizzler, and GULUserDefaults tests
-  Fixes Catalyst build options

The one failure is an unrelated Auth integration test problem with the Facebook API

#no-changelog

### CODE REVIEW

File: .github/workflows/appdistribution.yml [hunk 1]
```diff
@@ -43,6 +43,19 @@ jobs:
...
+    # Don't run on private repo unless it is a PR.
+    if: (github.repository == 'Firebase/firebase-ios-sdk' && github.event_name == 'schedule') || github.event_name == 'pull_request'
+
+    runs-on: macOS-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup Bundler
+      run: scripts/setup_bundler.sh
+    - name: Setup project and Build for Catalyst
+      # Only build the unit tests on Catalyst. Test stopped working when GHA moved to Xcode 11.4.1.
```
Comment (ryanwilson) [inline .github/workflows/appdistribution.yml:56]: Nit: out of date comment

File: GoogleUtilities/Tests/Unit/Network/GULNetworkTest.m [hunk 2]
```diff
@@ -17,6 +17,9 @@
 #import <XCTest/XCTest.h>
 #import "OCMock.h"
 
+#if !TARGET_OS_MACCATALYST
+// These tests are flaky on Catalyst. One of the tests typically fails.
```
Comment (ryanwilson) [inline GoogleUtilities/Tests/Unit/Network/GULNetworkTest.m:21]: Curious as to why that is, any ideas?

**Review by ryanwilson — APPROVED** [review]
Just the one nit and one question, thanks!

File: .github/workflows/appdistribution.yml (hunk 1 above)
Comment (paulb777) [reply .github/workflows/appdistribution.yml]: Deleted. Thanks!

File: GoogleUtilities/Tests/Unit/Network/GULNetworkTest.m (hunk 2 above)
Comment (paulb777) [reply GoogleUtilities/Tests/Unit/Network/GULNetworkTest.m:21]: Something to do with pending threads in tests crashing at completion, but I wasn't able to figure out. I'll push here to raise the Catalyst test baseline and leave these skips for a future task.

**Pull request merged**